.agents/skills/dsom-token-calculator/scripts/calculate-tokens.py

```
@@ -0,0 +1,140 @@
+        tok_ms = (time.perf_counter() - t1) * 1000
+
+        status = "[BLOCKED]" if total_tokens >= GATE_THRESHOLD else "[OK]    "
+        print(f"  {status} {total_tokens:>7,} tok  read={read_ms:>7.2f}ms  tok={tok_ms:>6.2f}ms  {filepath}")
+
+        if total_tokens >= GATE_THRESHOLD:
+            print(f"           [!] File exceeds {GATE_THRESHOLD:,}-token gate. Use view_file with line ranges instead of full load.")
+            if verbose_sections:
+                _report_sections(text, enc, os.path.basename(filepath))
+        return total_tokens, total_tokens >= GATE_THRESHOLD
+
+    except Exception as e:
```

Ruff (0.16.0)

[warning] Do not catch blind exception: `Exception`

(BLE001)